projects/packages/forms/src/blocks/field-checkbox/edit.js

```diff
@@ -58,6 +62,17 @@ export default function CheckboxFieldEdit( props ) {
 		[ setAttributes ]
 	);
 
+	const hasUpgradedToNewStyle = useRef( 1 );
+	// Ensure the className is set to 'is-style-list' if it is empty or not set.
+	// By updating the className on the second render, we can make sure that the block doesn't trigger a "Save" action.
+	useEffect( () => {
+		if ( className === '' || ( ! className && hasUpgradedToNewStyle.current === 2 ) ) {
+			__unstableMarkNextChangeAsNotPersistent();
+			setAttributes( { className: 'is-style-list' } );
+		}
+		hasUpgradedToNewStyle.current = 2;
+	}, [ className, setAttributes, __unstableMarkNextChangeAsNotPersistent ] ); // This effect is a placeholder for any future side effects.
+
```

Comment:
@enejb - can you leave a bit of explanation for why need the `__unstableMarkNextChangeAsNotPersistent` here?